### PR TITLE
Added hook for extension filters on category blog posts

### PR DESCRIPTION
**Pull request by assertchris**

### CODE REVIEW

**Comment by tractorcow** [comment]
Also see https://github.com/silverstripe/silverstripe-blog/pull/327

File: code/model/BlogCategory.php [hunk 1]
```diff
@@ -40,6 +40,17 @@ class BlogCategory extends DataObject implements CategorisationObject {
 	private static $extensions = array(
 		'URLSegmentExtension',
 	);
+	
+	/**
+	 * @return DataList
+	 */
+	public function BlogPosts() {
+		$blogPosts = parent::BlogPosts();
```
Comment (camfindlay) [inline code/model/BlogCategory.php]: this this not be simply $this->BlogPosts()? as there is a named relation here

Comment (assertchris) [inline code/model/BlogCategory.php]: I think calling `$this->method()` (every time) inside `function method` causes an infinite loop.

Comment (assertchris) [inline code/model/BlogCategory.php]: `parent::BlogPost()` does exist through `__call` on DataObject. It's pretty standard to override methods on child classes, and still call inherited functionality on parent classes. In this case, the inherited functionality is the code that generates the relationship DataList, and the overridden functionality is that which provides the hook.

Comment (tractorcow) [inline code/model/BlogCategory.php]: Yes I've used this before, and can confirm it falls back to `__call`

**Pull request merged**